src/FirstSteps.Tests/RPG/Adventures/MineTests.cs

```diff
@@ -23,19 +23,24 @@ public void Dwarf_Should_Gain_One_Coin_After_the_First_Visit__In_The_Mine()
         [InlineData(4, 2)]
         [InlineData(5, 3)]
         [InlineData(8, 4)]
-        [InlineData(9, 5)]
-        [InlineData(12, 6)]
-        public void Dwarf_Should_Gain_Three_Coins_When_Numbers_Are_Odd_Otherwise_Should_Gain_One_Coin(int expectedCoins, int visitCounter)
+        [InlineData(13, 5)]
```

Comment:
After the fifth visit hero should have 11 coins. 

According to the description:
- Every even visit: 3 coins
- Every visit divisible by five: **extra 2 coins to the normal amount**
- Thirteenth visit: always 0 coins
- Others: 1 coin

And this bold fragment is key.

I will paste here a fragment from my previous comment:

1. If your visit is the thirteenth visit return zero as soon as possible.
2. Calculate the number of coins based on the condition of an even visit.
3. If your visit is divisible by five **add two more coins to the result**.

The order is not accidental.